### ISSUE TITLE
Multiple focus rings on TableHeader

### ISSUE THREAD

**Issue by jluyau** [adobe/react-spectrum#4219]
# 🐛 Bug Report 

Under certain circumstances, using a mouse to click on a sortable TableView column, then switching modality to keyboard (by pressing Tab), multiple focus rings will show on the `TableHeader`.

## 🤔 Expected Behavior

Only a single focus ring should be displayed at a time

## 😯 Current Behavior

Multiple focus rings are being displayed.

## 🔦 Context

Having multiple focus rings makes it confusing to the user as to what element currently has focus.

Bug context:
For this bug to happen, TableView needs to have sortable and resizable columns. Currently the only way to reproduce the issue is by using `useAsyncList`'s `.reload()` function. In the code sandbox linked below, the `.reload()` function is tied to the `onSortChange` callback for simplicity. Current theory is that when the TableBody changes to a loading state then back to a settled state with items, that somehow messes with the next focusable child of TableHeader, causing the multiple focus rings on the header cells.

## 💻 Code Sample

https://codesandbox.io/s/muddy-microservice-vmoxen?file=/src/App.js

Repro steps:
1. Click `Age`/`Height`/`Weight` column header to sort by that column.
2. Wait until table has finished sorting
3. Press tab
4. Observer multiple focus rings.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.25.1
| Browser          | Chrome 110.0.5481.177
| Operating System | MacOS Ventura 13.2.1

## 🧢 Your Company/Team

Adobe/Quarry

**Comment by snowystinger** [comment]
I can't reproduce it in Chrome. Instead there is no focus ring. And in Safari, tab puts focus on the expected header.

I'm guessing part of this is that all items are cleared out, which means that focus goes to the collection as a whole, it won't be on the header cell anymore. Which is why tab brings you back to the header cell.

**Comment by jluyau** [comment]
odd. both @LFDanLu and i were able to repro in Chrome.

here's a screen recording:
https://user-images.githubusercontent.com/3901877/225781021-1b8dde80-a964-4a9c-81cf-655163dd3a7e.mov

Just tested in Firefox. Clicked Age to sort, then pressed tab. The last column (Short bio) gets the focus ring.

**Comment by snowystinger** [comment]
Ah, my mistake, I didn't notice it had to be one of the non-menu centric headers. I read the description
> TableView needs to have sortable and resizable columns

and jumped to the conclusion it was the first column i needed to sort. I can reproduce this now following the instructions more carefully.

**Comment by jluyau** [comment]
I think I isolated the problem some more. If `<TableView>` goes into a loading state when a column header is pressed, then returns back to a settled state, the multiple focus rings appear upon Tab press. 

Here's a storybook where pressing the column headers will go into the loading state, then go to a settled state 1 second later (via `setTimeout`): 
https://codesandbox.io/s/romantic-cherry-lqmmbq?file=/src/App.js

**Comment by jluyau** [comment]
In doing some more investigation, what was happening was that when the table was changing to a loading state (collection size == 0), the column headers were having their tabIndex removed. So if a user clicked a column header to sort a column (via server side sorting), the table will change to loading state `loading`, and the column headers would lose their tabIndex prop (moving focus to the window). Then when the table would repopulate with the sorted results, the focus would still be on the window and not on the column header as the user would expect (and pressing tab would have odd results, including multiple focus rings or normally untabbable elements being focused). 

Worked with @LFDanLu on a solution in the linked PR. If we keep focus on a column header that was the last selected key in the selection manager, then we don't observe any of the weird behavior. 

Another thing that came out of this investigation: do we want to keep the button in the resizable column header with tabIndex 0 or should it be -1 and change to 0 only when the column header is focused?
